src/components/action-popover/__internal__/action-popover-utils.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { ActionPopoverItem } from "../action-popover-item/action-popover-item.component";
+
+// Reusable type alias for item types
+type ReactItem = React.ReactChild | React.ReactFragment | React.ReactPortal;
+
+export const getItems = (
+  children: React.ReactNode | React.ReactNode[]
+): ReactItem[] =>
+  React.Children.toArray(children).filter(
+    (child) => React.isValidElement(child) && child.type === ActionPopoverItem
+  );
+
+export const isItemDisabled = (items: ReactItem[]) => (
+  value: number
+): boolean => {
+  const item = items[value];
+  return React.isValidElement(item) && !!item.props?.disabled;
+};
+
+export const findFirstFocusableItem = (
```

Comment:
Suggestion: If you make the change above you can remove the second parameter here and have something like:

```
const findFirstFocusableItem = (items: ReactItem[]): number => items.findIndex((_, index) => !isItemDisabled(items[index));

const findLastFocusableItem = (
  items: ReactItem[],
): number => {
  for (let i = items.length - 1; i >= 0; i--) {
    if (!isItemDisabled(items[i])) {
      return i;
    }
  }
  return -1;
};